crates/web/src/components/text_field.rs

```diff
@@ -0,0 +1,53 @@
+use std::collections::HashSet;
+
+use leptos::{component, create_memo, view, IntoView, MaybeProp, SignalGet};
+
+#[component]
+pub fn TextField(
+    #[prop(optional, into)] name: String,
+    #[prop(optional, into)] id: String,
+    #[prop(optional, into)] placeholder: String,
+    #[prop(optional, into)] value: String,
+    #[prop(optional, into)] label: String,
+    #[prop(optional, into, default = "text".to_string())] r#type: String,
```

Comment:
Can we have a `TextFieldVariant` enum for this?

Reply:
Sure!